internal/gencli/flag.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gencli
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/googleapis/gapic-generator-go/internal/pbinfo"
+)
+
+// Flag is used to represent fields as flags
+type Flag struct {
+	Name          string
+	Type          descriptor.FieldDescriptorProto_Type
+	Message       string
+	Repeated      bool
+	Required      bool
+	Usage         string
+	MessageImport pbinfo.ImportSpec
+	OneOfs        map[string]*Flag
+	IsOneOfField  bool
+	IsNested      bool
+}
+
+// GenOtherVarName generates the Go variable to store repeated Message & Enum string values
+func (f *Flag) GenOtherVarName(in string) string {
```

Comment:
Do these `Name` functions create variable names to be used internally by the implementation? If so, I don't want to be too picky about them. Perhaps because this PR is large, I'm having a hard time figuring out how they are used.

Reply:
Yes, they generate the names for variables used to store the `pflag` values in some cases and the final request payload message object always.

Example: `Kiosk.name` field might be the `--name` flag and have a containing variable `var KioskInputName string`.

Because these methods are used in the templates as helpers, I believe they have to be exported.